clients/adapter/src/main/java/org/eclipse/hono/adapter/client/command/DeviceConnectionClient.java

```diff
@@ -28,47 +26,7 @@
  * See Hono's <a href="https://www.eclipse.org/hono/docs/api/device-connection/">
  * Device Connection API specification</a> for a description of the result codes returned.
  */
-public interface DeviceConnectionClient extends Lifecycle {
-
-    /**
-     * Sets the given gateway as the last gateway that acted on behalf of the given device.
-     * <p>
-     * If a device connects directly instead of through a gateway, the device identifier itself is to be used as value
-     * for the <em>gatewayId</em> parameter.
-     *
-     * @param tenant The tenant that the device belongs to.
-     * @param deviceId The device id.
-     * @param gatewayId The gateway id (or the device id if the last message came from the device directly).
-     * @param context The currently active OpenTracing span context or {@code null} if no span is currently active.
-     *            An implementation should use this as the parent for any span it creates for tracing
-     *            the execution of this operation.
-     * @return A future indicating whether the operation succeeded or not.
-     * @throws NullPointerException if tenant, device id or gateway id are {@code null}.
-     */
-    Future<Void> setLastKnownGatewayForDevice(String tenant, String deviceId, String gatewayId, SpanContext context);
-
-    /**
-     * Gets the gateway that last acted on behalf of the given device.
-     * <p>
-     * If no last known gateway has been set for the given device yet, a failed future with status <em>Not Found</em>
-     * is returned.
-     *
-     * @param tenant The tenant that the device belongs to.
-     * @param deviceId The device id.
-     * @param context The currently active OpenTracing span context or {@code null} if no span is currently active.
-     *            An implementation should use this as the parent for any span it creates for tracing
-     *            the execution of this operation.
-     * @return A future indicating the result of the operation.
-     *         <p>
-     *         The future will succeed if a response with status 200 has been received from the device connection service.
-     *         In that case the value of the future will contain a <em>gateway-id</em> property with the
-     *         gateway id.
-     *         <p>
-     *         In case a status other then 200 is received, the future will fail with a
-     *         {@code org.eclipse.hono.client.ServiceInvocationException} containing the (error) status code returned by the service.
-     * @throws NullPointerException if tenant or device id are {@code null}.
-     */
-    Future<JsonObject> getLastKnownGatewayForDevice(String tenant, String deviceId, SpanContext context);
+public interface DeviceConnectionClient extends CommandRouterClient {
```

Comment:
I'm still not convinced that having the possibility to use just *one* bean (CommandRouterClient) for both the old commandConsumerFactory implementation and the new, CommandRouter-based one is worth the added confusion about the DeviceConnectionClient and CommandRouterClient being mixed up like this.
FMPOV, having a CommandRouterClient bean and a separate DeviceConnectionClient bean and allowing one or the other to be `null` would probably be easier to understand.

At least, when mixing the interfaces, I think it would be good to document the motivation in the DeviceConnectionClient javadoc.

Reply:
I'll give it another thought ...

Reply:
In any case we would have two different *CommandConsumerFactory* implementations, right?
1. The *old* one which simply wraps `org.eclipse.hono.client.ProtocolAdapterCommandConsumerFactory` and requires a *DeviceConnectionClient* and *DeviceRegistrationClient* implementation, and 
2- the new one which only requires the *CommandRouterClient*.

Based on whether the DeviceConnectionClient bean or the CommandRouterClient bean has been configured, we would instantiate the former or the latter, right?

Reply:
Yes.

Reply:
Ok, I have pushed corresponding changes ...